src/nix/flake.cc

```diff
@@ -1029,14 +1029,65 @@ struct CmdFlakeShow : FlakeCommand, MixJSON
                     auto name = visitor.getAttr(state->sName)->getString();
                     if (json) {
                         std::optional<std::string> description;
+                        std::optional<std::string> homepage;
+                        std::optional<std::vector<Bindings *>> maintainers;
+                        std::optional<std::vector<Symbol>> license;
                         if (auto aMeta = visitor.maybeGetAttr(state->sMeta)) {
                             if (auto aDescription = aMeta->maybeGetAttr(state->sDescription))
                                 description = aDescription->getString();
+                            if (auto aHomepage = aMeta->maybeGetAttr(state->sHomepage))
+                                homepage = aHomepage->getString();
+                            if (auto aMaintainers = aMeta->maybeGetAttr(state->sMaintainers))
+                                maintainers = aMaintainers->getListOfAttrs();
+                            if (auto aLicense = aMeta->maybeGetAttr(state->sLicense))
+                                license = aLicense->getAttrs();
```

Comment:
Couldn't you rather have a "toplevel" `std::optional<AttrCursor> aLicense` that you set here and use [below](https://github.com/NixOS/nix/pull/6828/files#diff-98db1e334292594945051880fac0aecb52c9206898bd3cbc79730d647b260824R1079-R1090)?

```suggestion
                            aLicense = aMeta->maybeGetAttr(state->sLicense):
```